### PR TITLE
Add BINARYEN_ROOT config

### PR DESCRIPTION
**Pull request by Zzzode**
Add the `BINARYEN_ROOT` path into `emcc --generate-config`

### CODE REVIEW

**Review by sbc100 — COMMENTED** [review]
lgtm % comments

File: tools/building.py [hunk 1]
```diff
@@ -1468,7 +1468,7 @@ def check_binaryen(bindir):
 
 def get_binaryen_bin():
   global binaryen_checked
-  rtn = os.path.join(config.BINARYEN_ROOT, 'bin')
```
[the remark this answers is not in the document]
Comment (Zzzode) [reply tools/building.py:1471]: Well, I found that in [emsdk_manifest.json](https://github.com/emscripten-core/emsdk/blob/main/emsdk_manifest.json), BINARYEN_ROOT has been set to `%installation_dir%`, which means `/path/to/upstream/` not `/path/to/upstream/bin`.
It seems that we should always add a `bin` in path.

[the remark this answers is not in the document]
Comment (sbc100) [reply tools/building.py:1471]: Yes, that is what I'm saying.   The `/bin` part needs to always be added onto the `BINARYEN_ROOT` which is why you can't remove this line.

[the remark this answers is not in the document]
Comment (sbc100) [reply tools/building.py:1471]: Note that this functions is explicitly setup to return the "bin" directory not the "root" directly of binaryen.

File: tools/config.py [hunk 2]
```diff
@@ -184,6 +184,9 @@ def generate_config(path):
   llvm_root = os.path.dirname(which('llvm-dis') or '/usr/bin/llvm-dis')
   config_data = config_data.replace('\'{{{ LLVM_ROOT }}}\'', repr(llvm_root))
 
+  binaryen_root = os.path.dirname(which('wasm-opt'))
```
[the remark this answers is not in the document]
Comment (Zzzode) [reply tools/config.py]: Thanks, I think `os.path.dirname(os.path.dirname(which('wasm-opt')))` could be useful.

**Comment by Zzzode** [comment]
It looks like `test_minimal_runtime_code_size_hello_webgl2_wasm2js` failed because of decreasing size(`total_output_size < total_expected_size`), but I don't know why, any more information?

**Comment by sbc100** [comment]
The code size test failures are unrelated.  They should be fixed if you rebase/merge with `main`.

**Review by sbc100 — APPROVED** [review]

**Pull request merged**